lib/matplotlib/projections/polar.py

```diff
@@ -679,20 +679,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.sticky_edges.y.append(0)
 
-    def _wrap_locator_formatter(self):
-        self.set_major_locator(RadialLocator(self.get_major_locator(),
-                                             self.axes))
-        self.isDefault_majloc = True
+    def set_major_locator(self, locator):
+        if not isinstance(locator, RadialLocator):
+            locator = RadialLocator(locator, self.axes)
```

Comment:
Minor side point (perhaps for @QuLogic, who introduced that API): perhaps RadialLocator can again lose the second parameter to its constructor (`axes`) and just access `self.axis.axes` instead?

Reply:
That's reasonable, but a separate topic. I've created #30296 to track it.